src/App.tsx

```diff
@@ -17,18 +70,28 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                handleFilterChange={handleFilterChange}
+                handleInputChange={handleInputChange}
+              />
             </div>
-
             <div className="block">
-              <Loader />
-              <TodoList />
+              {todos.length > 0 ? (
+                <TodoList
+                  todos={filteredTodos}
+                  selectedTodo={selectedTodo}
+                  setSelectedTodo={setSelectedTodo}
+                />
+              ) : (
+                <Loader />
+              )}
```

Comment:
You should show loader only in case you are loading something, if you get an empty array of todos a user can see the infinite loader